services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java

```diff
@@ -950,7 +950,11 @@ private Response performAccountLinking(AuthenticationSessionModel authSession, U
             }
         } else {
             this.session.users().addFederatedIdentity(this.realmModel, authenticatedUser, newModel);
+            federatedUser = session.users().getUserByFederatedIdentity(realmModel, newModel);
```

Comment:
```suggestion
            federatedUser = authenticatedUser;
```

Note that it should be same user (See line 927). So we can save some DB queries by not searching for this user again?